### PR TITLE
Add camera

### PR DESCRIPTION
**Pull request by JELoohuis**
- **Added support for the camera category of devices.**
    The camera device class is quite extensive, and some parts quite complex to integrate with Homey, so I stuck to those that could be readily tested with my device.
    
    Images are not yet supported due to Homey not having the correct rights in the Tuya API.
    It may be possible to decrypt the images sent in event messages, but this will require figuring out the exact encryption parameters used.
    The key for this is readily available through the Tuya API as local_key in the device details.
    
    Since the camera device class has a huge amount of capabilities,
    those that were deemed unlikely to be changed by hand often were implemented as settings.
    For those settings that may be changed semi-regularly flows were added.
    This is of course a compromise, as the settings that are not applicable cannot be disabled, unlike capabilities.
    
    The zoom control/stop and battery level are not supported by my test device, so they are untested.
    It would be nice to receive feedback on these from people with devices that do support them.

### CODE REVIEW

**Comment by WeeJeWel** [comment]
There was already a [branch for camera support](https://github.com/athombv/com.tuya/tree/feature/camera), but development has been paused because we cannot get access to the snapshot images or video streams.

Why make this PR at all, if we cannot get those?

File: .homeycompose/capabilities/ptz_control_horizontal.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      "title": {
+        "en": "Stop"
+      }
+    },
+    {
+      "id": "2",
+      "title": {
+        "en": "Right"
+      }
+    }
```
Comment (bobvandevijver) [inline .homeycompose/capabilities/ptz_control_horizontal.json:27]: We will need to use `up`/`idle`/`down` for the ternary component so the icons will be displayed correctly (also hold for the vertical and zoom control). Value translation will need to be done in the app code.

**Review by bobvandevijver — CHANGES_REQUESTED** [review]

File: drivers/camera/driver.flow.compose.json [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+                "en": "On"
+              }
+            }
+          ]
+        }
+      ]
+    }
+  ],
+  "conditions": [],
+  "triggers": [
```
[the remark this answers is not in the document]
Comment (JELoohuis) [reply drivers/camera/driver.flow.compose.json:159]: These are only ever set to true by events, and then to false by a timeout, so I don't think the false cards are very useful here. My test device did not generate multiple events within the timeout, but the cards may be useful if we want to combine events if this does happen.

File: drivers/camera/driver.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        [
+          "ptz_control",
+          "ptz_stop",
+          "zoom_control",
+          "zoom_stop",
+          "initiative_message",
+          "basic_private",
+          "wireless_electricity",
+        ]
+      ) {
```
[the remark this answers is not in the document]
Comment (JELoohuis) [reply drivers/camera/driver.js]: Fixed in 750408fd124b13de0682a5f926b66740f58c9ac6

File: drivers/camera/device.js [hunk 4]
```diff
@@ -0,0 +1,211 @@
+"use strict";
+
+const TuyaOAuth2Device = require("../../lib/TuyaOAuth2Device");
+const {
+  SIMPLE_CAMERA_CAPABILITIES,
+  CAMERA_ALARM_EVENT_CAPABILITIES,
+  CAMERA_SETTING_LABELS,
+} = require("./driver");
```
[the remark this answers is not in the document]
Comment (JELoohuis) [reply drivers/camera/device.js]: Fixed in 09a9c9fcdc7b838f441a3919afbea1b155620bf1

File: drivers/camera/driver.flow.compose.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (JELoohuis) [reply drivers/camera/driver.flow.compose.json:159]: I reworked the event handling in f30d99d79deea4a3904556bc7a68aa9f99df1adf and added the flows in ce1009080ffd531beca5bf010eab0a9b84e9e22e

**Review by bobvandevijver — APPROVED** [review]

**Pull request merged**